src/encoding/src/base64.cairo

```diff
@@ -25,10 +26,126 @@ impl Base64UrlEncoder of Encoder<Array<u8>> {
+    let mut result = array![];
+    if bytes.len() == 0 {
+        return result;
+    }
+    let mut p: u8 = 0;
+    let c = bytes.len() % 3;
+    if c == 1 {
+        p = 2;
+        bytes.append(0_u8);
+        bytes.append(0_u8);
+    } else if c == 2 {
+        p = 1;
+        bytes.append(0_u8);
+    }
+
+    let mut i = 0;
+    let bytes_len = bytes.len();
+    let last_iteration = bytes_len - 3;
+    loop {
+        if i == bytes_len {
+            break;
+        }
+        let n: u32 = (*bytes[i]).into()
+            * 65536_u32 | (*bytes[i + 1]).into()
+            * 256_u32 | (*bytes[i + 2]).into();
+        let e1: usize = ((n / 262144) & 63).try_into().unwrap();
+        let e2: usize = ((n / 4096) & 63).try_into().unwrap();
+        let e3: usize = ((n / 64) & 63).try_into().unwrap();
+        let e4: usize = (n & 63).try_into().unwrap();
+        result.append(*base64_chars[e1]);
+        result.append(*base64_chars[(e2)]);
+        if i == last_iteration {
+            if p == 2 {
+                result.append('=');
+                result.append('=');
+            } else if p == 1 {
+                result.append(*base64_chars[e3]);
+                result.append('=');
+            } else {
+                result.append(*base64_chars[e3]);
+                result.append(*base64_chars[e4]);
+            }
+        } else {
+            result.append(*base64_chars[e3]);
+            result.append(*base64_chars[e4]);
+        }
+        i += 3;
+    };
+    result
+}
+
+fn encode_felt(self: felt252, base64_chars: Array<u8>) -> Array<u8> {
+    let mut result = array![];
+
+    let mut num: u256 = self.into();
+    if num != 0 {
+        let (quotient, remainder) = DivRem::div_rem(
+            num, 65536_u256.try_into().expect('Division by 0')
+        );
+        let remainder: usize = remainder.try_into().unwrap();
+        let r3: usize = (remainder / 1024) & 63;
+        let r2: usize = (remainder / 16) & 63;
+        let r1: usize = (remainder * 4) & 63;
+        result.append(*base64_chars[r1]);
+        result.append(*base64_chars[r2]);
+        result.append(*base64_chars[r3]);
+        num = quotient;
+    }
+    loop {
+        if num == 0 {
+            break;
+        }
+        let (quotient, remainder) = DivRem::div_rem(
+            num, 16777216_u256.try_into().expect('Division by 0')
+        );
+        let remainder: usize = remainder.try_into().unwrap();
+        let r4: usize = remainder / 262144;
+        let r3: usize = (remainder / 4096) & 63;
+        let r2: usize = (remainder / 64) & 63;
+        let r1: usize = remainder & 63;
+        result.append(*base64_chars[r1]);
+        result.append(*base64_chars[r2]);
+        result.append(*base64_chars[r3]);
+        result.append(*base64_chars[r4]);
+        num = quotient;
+    };
+    loop {
+        if result.len() >= 43 {
+            break;
+        }
+        result.append('A');
+    };
+    result = result.reverse();
```

Comment:
why not construct the array backward to avoid this reverse ?

Reply:
The reasoning is the same as in ```bytes_be``` (see below) although I must confess I've not benchmarked this particular function with its reversed version.